dbos/_context.py

```diff
@@ -218,27 +232,38 @@ def _start_span(self, attributes: TracedAttributes) -> None:
         )
         attributes["authenticatedUserAssumedRole"] = self.assumed_role
         span = dbos_tracer.start_span(
-            attributes, parent=self.spans[-1] if len(self.spans) > 0 else None
+            attributes,
+            parent=self.context_spans[-1].span if len(self.context_spans) > 0 else None,
+        )
+        # Activate the current span
+        cm = use_span(
+            span,
+            end_on_exit=False,
+            record_exception=False,
+            set_status_on_exception=False,
```

Comment:
DBOS manually manages end_span and set status, so we set all of those options to false.